backend/src/server.ts

```diff
@@ -0,0 +1,34 @@
+import express from 'express';
+import cors from 'cors';
+import dotenv from 'dotenv';
+dotenv.config();
+import connectDB from '../config/db';
+import questionRoutes from './routes/questionRoutes';
+
+connectDB(); // Initialize MongoDB connection
+
+const PORT = process.env.PORT ?? 8080;
+
+const app = express();
+
+// Middleware for handling CORS and JSON parsing
+app.use(cors({
+  origin: 'http://localhost:3000',
```

Comment:
Vite default port is 5173? If we use 3000 it will fail on vite, right? Does ts cors accept wildcard? Propose to change to localhost:* if possible to avoid unnecessary pain, or add localhost:5173 at least

Reply:
Suryansh and I used postman for testing, which explains the port number. Will try wildcards, else will include both for testing and for integrating with frontend